### PR TITLE
Update ttCore.lua

### LINKED ISSUE #301: Needs to update to new GetMouseFocus API
### Describe the bug

```
77x TipTac/ttCore.lua:2777: attempt to call global 'GetMouseFocus' (a nil value)
[string "@TipTac/ttCore.lua"]:2777: in function `GetAnchorPosition'
[string "@TipTac/ttCore.lua"]:2903: in function `?'
[string "@TipTac/libs/LibFroznFunctions-1.0-24/LibFroznFunctions-1.0.lua"]:1175: in function `FireGroupEvent'
[string "@TipTac/ttCore.lua"]:1736: in function `SetCurrentDisplayParams'
[string "@TipTac/ttCore.lua"]:1621: in function <TipTac/ttCore.lua:1620>
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "=[C]"]: in function `Show'
[string "@Blizzard_Settings_Shared/Blizzard_SettingControls.lua"]:38: in function <...lizzard_Settings_Shared/Blizzard_SettingControls.lua:26>

Locals:
self = TipTac {
 RightEdge = Texture {
 }
 TopEdge = Texture {
 }
 BottomRightCorner = Texture {
 }
 LeftEdge = Texture {
 }
 BottomEdge = Texture {
 }
 BottomLeftCorner = Texture {
 }
 text = FontString {
 }
 close = Button {
 }
 TopLeftCorner = Texture {
 }
 Center = Texture {
 }
 backdropInfo = <table> {
 }
 TopRightCorner = Texture {
 }
}
tip = SettingsTooltip {
 textLeft1Font = "GameTooltipHeaderText"
 RightEdge = Texture {
 }
 TextRight1 = SettingsTooltipTextRight1 {
 }
 TopOverlay = Texture {
 }
 BottomLeftCorner = Texture {
 }
 TextLeft1 = SettingsTooltipTextLeft1 {
 }
 Center = Texture {
 }
 textRight2Font = "GameTooltipText"
 BottomOverlay = Texture {
 }
 BottomEdge = Texture {
 }
 textRight1Font = "GameTooltipHeaderText"
 BottomRightCorner = Texture {
 }
 NineSlice = Frame {
 }
 textLeft2Font = "GameTooltipText"
 TopEdge = Texture {
 }
 TextRight2 = SettingsTooltipTextRight2 {
 }
 TextLeft2 = SettingsTooltipTextLeft2 {
 }
 TopRightCorner = Texture {
 }
 TopLeftCorner = Texture {
 }
 backdropInfo = <table> {
 }
 LeftEdge = Texture {
 }
}
frameParams = <table> {
 gradient = Texture {
 }
 config = <table> {
 }
 currentDisplayParams = <table> {
 }
 frameName = "SettingsTooltip"
}
isUnit = nil
(*temporary) = nil
(*temporary) = "attempt to call global 'GetMouseFocus' (a nil value)"
TT_CacheForFrames = <table> {
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
  = <table> {
 }
}
TT_TIP_CONTENT = <table> {
 unknownOnCleared = 8
 spell = 3
 item = 4
 aura = 2
 unknownOnShow = 7
 others = 6
 action = 5
 unit = 1
}
cfg = <table> {
 showUnitTip = true
 classification_elite = "+%s "
 barHeight = 10
 if_showCurrencyId = true
 showMount = false
 if_showKeystoneTimeLimit = false
 if_showAuraSpellIdAndRank = true
 anchorFrameUnitPoint = "BOTTOMRIGHT"
 if_flyoutColoredBorder = false
 barTexture = "Interface\AddOns\Forte_Core\Textures\Smooth"
 if_iconAnchor = "TOPLEFT"
 if_runeforgePowerColoredBorder = false
 if_transmogSetQualityBorder = false
 classification_rare = "%s|r (Rare) "
 left = 1442.000610
 t_talentFormat = 2
 colorGuild = <table> {
 }
 hideAllTipsInCombat = true
 classification_trivial = "%s "
 if_petActionColoredBorder = false
 if_battlePetAbilityColoredBorder = false
 if_transmogIllusionColoredBorder = false
 if_showNpcId = false
 targetYouText = "<<YOU>>"
 if_conduitQualityBorder = false
 top = 628.999939
 if_transmogAppearanceItemQualityBorder = false
 showCurrentUnitSpeed = false
 if_currencyQualityBorder = false
 if_enable = true
 minimapConfig = <table> {
 }
 if_achievmentColoredBorder = false
 anchorFrameTipPoint = "BOTTOMRIGHT"
 if_showSpellIdAndRank = true
 colorSameGuild = <table> {
 }
 if_smartIcons = false
 showGuildRank = false
 showHiddenTipsOnShift = false
 if_iconOffsetX = -40
 if_auraSpellColoredBorder = false
 healthBar = false
 if_battlePetQualityBorder = false
 if_modifyKeystoneTips = false
 optionsBottom = 493.999542
 classification_normal = "%s "
 if_spellColoredBorder = false
 if_showItemId = true
 enableAuras = false
 if_showIcon = false
 if_showAuraCaster = true
 classification_minus = "%s"
 showDebuffs = false
 if_iconTooltipAnchor = "TOPLEFT"
 if_itemQualityBorder = false
 if_showKeystoneAffixInfo = false
 enableBars = false
 if_showItemLevelAndId = true
 colSameGuild = "|cff00ffff"
 colGuild = "|cff00ff00"
 showBuffs = false
 classColoredBorder = false
 colorNameByClass = true
 anchorWorldTipPoint = "BOTTOMRIGHT"
 classification_rareelite = "+%s|r (Rare) "
 anchorFrameTipType = "mouse"
 if_iconOffsetY = -1
 classification_worldboss = "%s|r (Boss) "
 iconAnchor = "TOPLEFT"
 anchorWorldUnitPoint = "BOTTOMRIGHT"
 showMountIcon = true
 showMinimapIcon = f
```

Addon should be using the new `GetMouseFoci` API instead

### Steps to Reproduce

Load the addon, it crashes immediately and repeatedly

### Expected behavior

It to not crash on load :)

### Additional context

TY for maintaining this, it's by far and away my favorite tooltip addon :)

### Version of TipTac Reborn

24.05.26

### Flavor/Version of WoW

Latest Retail

### PR DESCRIPTION
**Pull request by darkpsy3934**
Resolves #301 - Changed GetMouseFocus to GetMouseFoci

### CODE REVIEW

File: TipTac/ttCore.lua [hunk 1]
```diff
@@ -2811,7 +2811,7 @@ function tt:GetAnchorPosition(tip)
 		end
 	end
 	
-	local mouseFocus = GetMouseFocus();
+	local mouseFocus = GetMouseFoci();
```
Comment (romracer) [inline TipTac/ttCore.lua:2814]: GetMouseFoci() returns a table.  I'm not sure this is right.  Probably should be something like:

```
local mouseFocus
if GetMouseFocus then
  mouseFocus = GetMouseFocus()
elseif GetMouseFoci then
  local foci = GetMouseFoci()
  mouseFocus = foci[1] or nil
end
```

but that might be better as a function in LibFroznFunctions:

```
function LibFroznFunctions:GetMouseFocus()
    if GetMouseFocus then
        return GetMouseFocus()
    elseif GetMouseFoci then
        local foci = GetMouseFoci()
        return foci[1] or nil
    end

    return nil
end
```

and then called like:

```
local mouseFocus = LibFroznFunctions:GetMouseFocus()
```

File: TipTac/ttCore.lua [hunk 2]
```diff
@@ -3008,7 +3008,7 @@ function tt:SetUnitRecordFromTip(tip)
 	-- and it will return as "mouseover", but the "mouseover" unit id is still invalid at this point for those unitframes!
 	-- to overcome this problem, we look if the mouse is over a unitframe, and if that unitframe has a unit attribute set?
 	if (not unitID) then
-		local mouseFocus = GetMouseFocus();
+		local mouseFocus = GetMouseFoci();
```
Comment (romracer) [inline TipTac/ttCore.lua:3011]: Same as my previous comment.

**Review by romracer — CHANGES_REQUESTED** [review]

**Comment by darkpsy3934** [comment]
When I made the change GetMouseFoci(), the BugTracker errors did disappear. Unless I just didn't encounter anything where the error would appear again?

**Comment by darkpsy3934** [comment]
Ah, I see what's happening. Since GetMouseFocus() is still in previous versions of the game pre-11.0, it would need to be called instead of GetMouseFoci(). Otherwise an error would be thrown.

I'll make the change and commit again!

**Comment by romracer** [comment]
The error would stop, because the code is no longer calling a non-existent API function (`GetMouseFocus()`).  But code further down no longer works, because mouseFocus isn't what those lines of code think it is anymore.  It's table, so for example on line 3013 (before the changes), the check for a `unit` attribute won't succeed, because tables don't have that.  But the top most frame (at index 1 in the table), _might_ have that if its a unit.

**Comment by darkpsy3934** [comment]
Alright, that should be good now!

**Comment by darkpsy3934** [comment]
Alright, testing this out it appears the item quality colours on the borders of the tooltip are no longer appearing. No errors are being thrown, however.

**Comment by darkpsy3934** [comment]
I'm just going to close this out for now. I'll submit a new PR when I've got it worked out :)

**Comment by romracer** [comment]
Try #304 if you want

**Pull request closed without merging**